sw/device/lib/dif/register.h

```diff
@@ -0,0 +1,49 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_DIF_REGISTER_H_
+#define SW_DEVICE_LIB_DIF_REGISTER_H_
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+typedef volatile uint32_t *reg32_t;
+
+inline reg32_t reg32_new(uintptr_t address) { return (reg32_t)address; }
```

Comment:
Hmm.  I don't like this.  To me `_new` has an allocate-y smell to it, or at the very least complex-initialization-y smell.

Reply:
Do you like `_from_addr`?

Reply:
I quite like `_from_addr`, I think the name is descriptive.